src/transforms/sample/transform.rs

```diff
@@ -1,119 +1,32 @@
-    /// sampled.
-    ///
-    /// Each unique value for the key creates a bucket of related events to be sampled together
-    /// and the rate is applied to the buckets themselves to sample `1/N` buckets.  The overall rate
-    /// of sampling may differ from the configured one if values in the field are not uniformly
-    /// distributed. If left unspecified, or if the event doesn’t have `key_field`, then the
-    /// event is sampled independently.
-    ///
-    /// This can be useful to, for example, ensure that all logs for a given transaction are
-    /// sampled together, but that overall `1/N` transactions are sampled.
-    #[configurable(metadata(docs::examples = "message",))]
-    pub key_field: Option<String>,
-
-    /// A logical condition used to exclude events from sampling.
-    pub exclude: Option<AnyCondition>,
-}
-
-impl GenerateConfig for SampleConfig {
-    fn generate_config() -> toml::Value {
-        toml::Value::try_from(Self {
-            rate: 10,
-            key_field: None,
-            exclude: None::<AnyCondition>,
-        })
-        .unwrap()
-    }
-}
-
-#[async_trait::async_trait]
-#[typetag::serde(name = "sample")]
-impl TransformConfig for SampleConfig {
-    async fn build(&self, context: &TransformContext) -> crate::Result<Transform> {
-        Ok(Transform::function(Sample::new(
-            self.rate,
-            self.key_field.clone(),
-            self.exclude
-                .as_ref()
-                .map(|condition| condition.build(&context.enrichment_tables))
-                .transpose()?,
-        )))
-    }
-
-    fn input(&self) -> Input {
-        Input::new(DataType::Log | DataType::Trace)
-    }
-
-    fn outputs(
-        &self,
-        _: vector_lib::enrichment::TableRegistry,
-        input_definitions: &[(OutputId, schema::Definition)],
-        _: LogNamespace,
-    ) -> Vec<TransformOutput> {
-        vec![TransformOutput::new(
-            DataType::Log | DataType::Trace,
-            input_definitions
-                .iter()
-                .map(|(output, definition)| {
-                    (
-                        output.clone(),
-                        definition.clone().with_source_metadata(
-                            SampleConfig::NAME,
-                            Some(LegacyKey::Overwrite(owned_value_path!("sample_rate"))),
-                            &owned_value_path!("sample_rate"),
-                            Kind::bytes(),
-                            None,
-                        ),
-                    )
-                })
-                .collect(),
-        )]
-    }
-}
-
 #[derive(Clone)]
 pub struct Sample {
+    name: String,
     rate: u64,
     key_field: Option<String>,
     exclude: Option<Condition>,
     count: u64,
 }
 
 impl Sample {
-    pub const fn new(rate: u64, key_field: Option<String>, exclude: Option<Condition>) -> Self {
+    #![allow(dead_code)]
```

Comment:
Does clippy complain about this?

Reply:
removed in https://github.com/vectordotdev/vector/pull/19806/commits/56e68b935163c92f534416d47d31af2f5edab71d